enhancements/olm/report-sdk-operators.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Enable tracking of SDK operators in OLM metrics
+authors:
+  - "@varshaprasad96"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-05-28
+status: implementable
+---
+
+# Enable tracking of SDK operators in OLM metrics
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+This enhancement proposes the addition of operator specific labels to existing OLM metrics, in order to collect data on Redhat operators which utilize SDK to build image bundle.
+
+## Motivation
+
+Collecting metrics regarding the usage of Operator SDK among operator authors is important to measure the reachability of the tool. Since OLM already has a pipeline built to obtain metrics on operators and report it to the RedHat OpenShift telemetry, we intend to utilize the same and obtain data on Red Hat operators (installed in OpenShift namespaces) using SDK to build bundle images.
```

Comment:
Why is this data not captured during the certification pipeline?  From a given cluster, an operator and a version uniquely identify an operator's binary bits that can be checked out of band.  I don't think clusters need to report how those operators are built - that would be like a metric that reports what base image a given red hat image uses - it's trivially reproducible from outside of telemetry, and so should not be part of the metrics stack without a very high value use case.

Reply:
I understood the goal to be identification of operators built via sdk outside of red hat pipelines as well as in, to gauge user adoption.

Reply:
Those operators must be submitted by someone to operator hub, right?  I think metrics are fine to be reported from the operator, but I don't think we'd need to or want to gather that from telemetry today.  I might prefer the operator metadata include that data as part of the build process, and then OLM summarize a single metric that reports the count of installed operators with that metadata (instead of asking the operator itself).

Reply:
Correct to be clear we are just implenting labels so that we can pull this data from OLM with this in mind can we please get this PR approved?